docs/puzzles/day21.md

```diff
@@ -1,11 +1,312 @@
 import Solver from "../../../../website/src/components/Solver.js"
 
-# Day 21
+# Day 21: Dirac Dice
+by @sjrd
 
 ## Puzzle description
 
 https://adventofcode.com/2021/day/21
 
+## Modeling and parsing the input
+
+There are no more surprises here, so let us directly see the code:
+
+```scala
+type Cell = Int // from 0 to 9, to simplify computations
+
+case class Player(cell: Cell, score: Long)
+
+type Players = (Player, Player)
+
+def parseInput(input: String): Players =
+  val lines = input.split("\n")
+  (parsePlayer(lines(0)), parsePlayer(lines(1)))
+
+def parsePlayer(line: String): Player =
+  line match
+    case s"Player $num starting position: $cell" =>
+      Player(cell.toInt - 1, 0L)
+```
+
+The only thing worth nothing is that we use numbers 0 to 9 for the cells (the "spaces") instead of 1 to 10.
+The only reason is that it simplifies computations for wrapping around.
+
+## The deterministic die
+
+For the first play with the deterministic die, we have to return the score of the losing player, while keeping tabs on how many times the die was rolled.
+We model the deterministic die as an instance of a class `DeterministicDie`:
+
+```scala
+final class DeterministicDie {
+  var throwCount: Int = 0
```

Comment:
realizing now that the “setter” should probably be private, and only the “getter” should be public. But you might say that it’d be overengineered.

Reply:
I thought about that but I self-concluded it would be overengineered, indeed.